core/src/main/scala/com/criteo/cuttle/App.scala

```diff
@@ -144,7 +143,26 @@ private[cuttle] case class App[S <: Scheduling](project: CuttleProject[S],
   import App._
   import project.{scheduler, workflow}
 
-  private def allJobs = workflow.vertices.map(_.id)
+  private val allJobs = workflow.vertices.map(_.id)
+
+  private def processOnJobIds[T](jobsQueryString: String)(process: Set[String] => T): T =
```

Comment:
What you did works & does factor some code. 

Though, I think it is preferrable to have something like :
```scala
def parseJobIds(jobsQueryString: String): Set[String] = ???
def getJobsOrNotFound(jobsQueryString: String): Either[Response, Set[Job[S]]] = ???
```
Then use 
```scala
val jobnames = parseJobIds(jobsQueryString) 
```
and
```scala 
case GET at url"/.../?jobs=$jobs" => 
  getJobsOrNotFound.fold(identity, jobNames => {...})
```

This approach :
- creates a simplistic, testable and reusable function getJobsIds (I do not see any good reason why processOnJobIds should require the argument _process_ )
- introduces (minor) complexity where the code is called as opposed to the parsing / lookup functions
- Is, _subjectively_, more readable

I would agree with your approach if you did not have to pass jobs explicitly.